tests/test_ingestion.py

```diff
@@ -0,0 +1,183 @@
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+
+import os
+import shutil
+import tempfile
+import unittest
+
+import lsst.utils.tests
+import lsst.pex.exceptions as pexExcept
+import lsst.daf.persistence as dafPersist
+import lsst.ap.verify.ingestion as ingestion
+
+
+# TODO: convert test data to obs_test in DM-13849
+# In the meantime, run scons -j or pytest --forked
+class IngestionTestSuite(lsst.utils.tests.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            cls.testData = lsst.utils.getPackageDir("testdata_decam")
+        except pexExcept.NotFoundError:
+            message = "testdata_decam not setup. Skipping."
+            raise unittest.SkipTest(message)
+
+        cls.testApVerifyData = os.path.join('tests', 'ingestion')
+        cls.rawDataId = {'visit': 229388, 'ccdnum': 1}
+        # TODO: butler queries fail without this extra info; related to DM-12672?
+        cls.calibDataId = {'visit': 229388, 'ccdnum': 1, 'filter': 'z', 'date': '2013-09-01'}
+        cls.defectDataId = {'path': os.path.join('defects', 'D_n20150105t0115_c23_r2134p01_bpm.fits')}
+
+    def setUp(self):
+        self._repo = tempfile.mkdtemp()
+        self._calibRepo = os.path.join(self._repo, 'calibs')
+        templateRepo = os.path.join(IngestionTestSuite.testApVerifyData, 'repoTemplate')
+
+        # Initialize as a valid repository
+        # Can't call copytree on (templateRepo, self._repo) because latter already exists
+        testFiles = os.listdir(templateRepo)
+        for testFile in testFiles:
+            original = os.path.join(templateRepo, testFile)
+            copy = os.path.join(self._repo, testFile)
+            if os.path.isdir(original):
+                shutil.copytree(original, copy)
+            else:
+                shutil.copy2(original, copy)
+
+        # Initialize calib repo
+        # Making the directory appears to be both necessary and sufficient
+        os.mkdir(self._calibRepo)
+
+    def _rawButler(self):
+        """Return multiple ways of querying calibration repositories.
+
+        Returns
+        -------
+        butler : `lsst.daf.persistence.Butler`
+            A butler that should all be capable of finding ingested science data.
```

Comment:
Here and in the next one, "A butler that should all be capable of finding" is oddly worded. Do you mean "should be capable of finding all ingested [...] data" ?